cpp/src/strings/convert/convert_hex.cu

```diff
@@ -0,0 +1,163 @@
+        {
+            char ch = str[index];
+            if( ch >= '0' && ch <= '9' )
+            {
+                result += static_cast<int64_t>(ch-48) * base;
+                base <<= 4; // multiply by 16
+            }
+            else if( ch >= 'A' && ch <= 'Z' )
+            {
+                result += static_cast<int64_t>(ch-55) * base;
+                base <<= 4; // multiply by 16
+            }
+            else if( ch >= 'a' && ch <= 'z' )
+            {
+                result += static_cast<int64_t>(ch-87) * base;
+                base <<= 4; // multiply by 16
+            }
+        }
+        return result;
+    }
+
+    __device__ IntegerType operator()(size_type idx)
+    {
+        if( strings_column.is_null(idx) )
+            return static_cast<IntegerType>(0);
+        // the cast to IntegerType will create predictable results
+        // for integers that are larger than the IntegerType can hold
+        return static_cast<IntegerType>(string_to_integer(strings_column.element<string_view>(idx)));
+    }
+};
+
+/**
+ * @brief The dispatch functions for converting strings to integers.
+ *
+ * The output_column is expected to be one of the integer types only.
+ */
+struct dispatch_hex_to_integers_fn
+{
+    template <typename IntegerType, std::enable_if_t<std::is_integral<IntegerType>::value>* = nullptr>
+    void operator()( column_device_view const& strings_column,
+                     mutable_column_view& output_column,
+                     cudaStream_t stream ) const
+    {
+        auto d_results = output_column.data<IntegerType>();
+        thrust::transform( rmm::exec_policy(stream)->on(stream),
+            thrust::make_counting_iterator<size_type>(0),
+            thrust::make_counting_iterator<size_type>(strings_column.size()),
+            d_results, hex_to_integer_fn<IntegerType>{strings_column});
+    }
+    // non-integral types throw an exception
+    template <typename T, std::enable_if_t<not std::is_integral<T>::value>* = nullptr>
+    void operator()(column_device_view const&, mutable_column_view&, cudaStream_t) const
+    {
+        CUDF_FAIL("Output for hex_to_integers must be an integral type.");
+    }
+};
+
+template <>
+void dispatch_hex_to_integers_fn::operator()<experimental::bool8>(column_device_view const&, mutable_column_view&, cudaStream_t) const
+{
+    CUDF_FAIL("Output for hex_to_integers must not be a boolean type.");
+}
+
+} // namespace
+
+
+// This will convert a strings column into any integer column type.
+std::unique_ptr<column> hex_to_integers( strings_column_view const& strings,
+                                         data_type output_type,
+                                         rmm::mr::device_memory_resource* mr = rmm::mr::get_default_resource(),
+                                         cudaStream_t stream = 0)
+{
+    size_type strings_count = strings.size();
+    if( strings_count == 0 )
+        return make_numeric_column( output_type, 0 );
```

Comment:
I think empty_like would be more cudf-idiomatic.

Reply:
How about `make_empty_column` instead?
https://github.com/rapidsai/cudf/blob/5a73e75be813e76f750f3ad1416d8cdc54906bbd/cpp/include/cudf/column/column_factories.hpp#L36